### PR TITLE
Initial Commit :: Sonatype Nexus 2.x

### PR DESCRIPTION
**Pull request by m4rkmckenna**

### CODE REVIEW

File: catalog/nexus/nexus.bom [hunk 1]
```diff
@@ -0,0 +1,88 @@
+brooklyn.catalog:
+  version: &nexusVersion "1.0.0-SNAPSHOT" # BROOKLYN_NEXUS_VERSION
+  iconUrl: &nexusIconUrl "https://github.com/sonatype.png"
+  name: &nexusName "Nexus Repository OSS"
```
Comment (tbouron) [inline catalog/nexus/nexus.bom]: Would be great to directly include the [`publish`](https://github.com/brooklyncentral/blueprint-repository) block to release on blueprint.cloudsoft.io

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.brooklyncentral.nexus</groupId>
+    <artifactId>brooklyn-nexus</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+
+
+</project>
```
Comment (tbouron) [inline pom.xml]: Do we need the `pom.xml` at all?

File: catalog/nexus/nexus.bom [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  description: &nexusDescription |
+    The world's only repository manager with FREE support for popular formats.
+  items:
+    - id: nexus
+      name: *nexusName
+      version: *nexusVersion
+      iconUrl: *nexusIconUrl
+      description: *nexusDescription
+      itemType: entity
```
Comment (tbouron) [inline catalog/nexus/nexus.bom]: Wouldn't be better to be a `template`?

**Review by tbouron — DISMISSED** [review]
Tested on AWS, SoftLayer and GCE successfully, LGTM!

Only minor comments.

File: catalog/nexus/nexus.bom (hunk 1 above)
Comment (m4rkmckenna) [reply catalog/nexus/nexus.bom]: I have this in a follow up commit

File: catalog/nexus/nexus.bom (hunk 3 above)
Comment (m4rkmckenna) [reply catalog/nexus/nexus.bom]: Disagree

File: pom.xml (hunk 2 above)
Comment (m4rkmckenna) [reply pom.xml]: We do if you want to package this bom correctly like clocker

File: catalog/nexus/nexus.bom [hunk 4]
```diff
@@ -0,0 +1,91 @@
+brooklyn.catalog:
+  publish: &nexusPublish
```
Comment (tbouron) [inline catalog/nexus/nexus.bom:2]: Do we need the `&nexusPublish`

File: catalog/nexus/nexus.bom [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      publish: *nexusPublish
+      itemType: entity
+      item:
+        type: org.apache.brooklyn.entity.software.base.VanillaSoftwareProcess
+        name: "Nexus"
+        id: nexus
+        provisioning.properties:
+          osFamily: centos
+          osVersionRegex: 7
+        brooklyn.parameters: &nexusParameters
```
Comment (tbouron) [inline catalog/nexus/nexus.bom]: Do we need the `&nexusParameters`

File: catalog/nexus/nexus.bom (hunk 3 above)
Comment (tbouron) [reply catalog/nexus/nexus.bom]: Why is that?

File: catalog/nexus/nexus.bom (hunk 4 above)
Comment (m4rkmckenna) [reply catalog/nexus/nexus.bom:2]: Its used below

File: catalog/nexus/nexus.bom [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      description: *nexusDescription
+      publish: *nexusPublish
+      itemType: entity
+      item:
+        type: org.apache.brooklyn.entity.software.base.VanillaSoftwareProcess
+        name: "Nexus"
+        id: nexus
+        provisioning.properties:
+          osFamily: centos
+          osVersionRegex: 7
```
Comment (drigodwin) [inline catalog/nexus/nexus.bom:23]: Why not use the common utils centos entity?

**Review by drigodwin — APPROVED** [review]
Looks good, thanks @m4rkmckenna

**Pull request merged**